CHANGELOG.md

```diff
@@ -1,83 +1,94 @@
-- [Removed] Archive no longer needs an RPC url to function
-- [Changed] Print out the Postgres URL at startup
-- [Added] `max_block_load` to configure the maximum number of blocks loaded at once
-- [Added] Two new options to the configuration relative to state tracing:
-  - `targets` for specifying runtime targets to trace in WASM
-  - `folder` where WASM blobs with tracing enabled are kept.
-  - More on state-tracing [here](https://github.com/paritytech/substrate-archive/wiki/6.\)-State-Tracing-&-Balance-Reconciliation)
-- [Changed] Archive config is now separated into sections for readability. Migration is manual but looking at the new `archive.conf` in `polkadot-archive` or `node-template-archive` folders should help.
-
-#### Internal Changes
-
-- [Changed] Postgres SQL queries are now type checked
-- [QoL] Refactor file layout to `substrate-archive` and `substrate-archive-backend`.
-- [perf] Decouple Database actors
-- [QoL] upgrade to SQLx 0.5.0
-- [perf] Overhaul of block indexing. Now uses a Iterator to only collect batches of blocks from the database,
-taking advantage of the better reading performance of sequential data access. Gathering blocks by RPC is no longer done.
-- [perf] a new module `runtime_version_cache` is introduced in order to cache and run a binary search on runtime version & blocks.
-- [perf] better queries for the set difference between the storage and blocks table - makes querying for missing storage more efficient
-- [err] Better handling of SQL errors
-- [perf] switch to a leaner, 'lower-level' actor framework (xtra)
-- [perf] switch to a background-task-queue for executing blocks. This uses significantly less memory and
-  persists blocks that need to be executed on-disk.
-- [QoL] remove the last frame dependency, `frame-system`. Archive now relies only on generic traits defined in substrate-core.
-- new method of batch inserts avoids starving the Postgres Pool of connections
-- [perf] Notification stream from Postgres indexes storage changes in the background.
-- [Changed] Switch substrate-archive `LocalCallExecutor` with Substrate's `LocalCallExecutor`
-
-### Polkadot Archive
-
-- [Changed] Config file is now optional. Can configure polkadot archive entirely through environment variables.
-  - the environment variables that need to be set are `CHAIN_DATA_DB` and `DATABASE_URL`.
-- [Changed] Polkadot archive will archive `polkadot` by default if the `--chain` CLI option is not passed.
-- [Changed] remove `rpc_url` from the polkadot-archive TOML configuration file
-- [Changed] All options in config file apart from `db_url`.
-
-## [v0.4.0]
+# Changelog
+
+All notable changes for substrate-archive will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
+and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+## [Unreleased]
+## [v0.5.2] - 2021-06-02
+### Added
+- Tracing Wasm Runtimes `v0.9.0`, `v0.9.1`, `v0.9.2`, `v0.9.3`
```

Comment:
I don't know if it's mandatory, but it's common to always include a reference to the PRs in the CHANGELOG. Not a blocker, but it is my preferred way. I didn't do this in the previous version because there weren't really any PRs associated with the change, so not a blocker. But going forward I think we should try to add PR links whenever it makes sense.

Reply:
@insipx worth mentioning that in this version the chain spec is mandatory? And does that mean this should be `v0.6` (breaking)?

Reply:
> I don't know if it's mandatory, but it's common to always include a reference to the PRs in the CHANGELOG. Not a blocker, but it is my preferred way. I didn't do this in the previous version because there weren't really any PRs associated with the change, so not a blocker. But going forward I think we should try to add PR links whenever it makes sense.

Yea I agree, it's a nice reference to have.

Reply:
Chain spec actually not mandatory, decided against it because it still didn't seem necessary to me, since derivative implementations (i.e polkadot-archive) can make them required if necessary

Reply:
Awesome, added the links.